src/seismometer/controls/selection.py

```diff
@@ -39,57 +39,60 @@ def __init__(self, title: str, options: tuple[str], value: tuple[str] = None, sh
                 tooltip=str(option),
                 disabled=False,
                 button_style="",
-                layout=Layout(width="90%"),
+                layout=Layout(flex="1 0 auto"),
             )
             sub_toggle.observe(self._on_button_change, "value")
-            self.buttons.append(sub_toggle)
-            self.value_from_button[sub_toggle] = option
+            self.button_from_option[option] = sub_toggle
+            self.option_from_button[sub_toggle] = option
 
-        self.layout = Layout(width="max-content", min_width="100px")
-        self.label = Label(title)
-        if show_title:
-            self.children = [self.label] + self.buttons
+        self.layout = Layout(width="max-content", min_width="var(--jp-widgets-inline-label-width)")
+        self.button_box = VBox(
+            children=[button for option, button in self.button_from_option.items()],
+            layout=Layout(max_height="calc(7* var(--jp-widgets-inline-height))", align_items="flex-start"),
+        )
+        if title:
+            self.title_label = Label(title)
+            self.children = [self.title_label] + [self.button_box]
         else:
-            self.children = self.buttons
-        self._on_button_change()
+            self.title_label = None
+            self.children = [self.button_box]
+        self.observe(self._on_value_change, "value")
+        self._disabled = False
+
+    @property
+    def disabled(self) -> bool:
+        return self._disabled
+
+    @disabled.setter
+    def disabled(self, disabled: bool):
+        self._disabled = disabled
+        for widget in self.button_from_option.values():
+            widget.disabled = disabled
 
     def _on_button_change(self, change=None):
         """Bubble down control change."""
         if change:
             change["owner"].icon = "check" if change["owner"].value else ""
-        self.value = self.control_value
-
-    @property
-    def title(self) -> str:
-        """Selection title."""
-        return self.label.value
-
-    @title.setter
-    def title(self, title: str):
-        self.label.value = title
-
-    @property
-    def control_value(self) -> tuple:
-        """Value from child widgets."""
-        return tuple(self.value_from_button[x] for x in self.buttons if x.value)
+        self.value = tuple(option for button, option in self.option_from_button.items() if button.value)
 
-    @control_value.setter
-    def control_value(self, new_value: tuple[str]):
+    def _on_value_change(self, change=None):
```

Comment:
Q: parameter `change` isn't used in this function: what is it's purpose

Reply:
this is provided by traitlets, it's a dictionary that lets you know what was changed. It has `change["owner"]`, `change["new"]`, and `change["old"]` so you can decided what to do if you want fancy handling.